packages/explorer/src/app/(explorer)/[chainName]/worlds/WorldsForm.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+
+import Image from "next/image";
+import { useParams, useRouter } from "next/navigation";
+import { Address, isAddress } from "viem";
+import * as z from "zod";
+import { Command as CommandPrimitive } from "cmdk";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Button } from "../../../../components/ui/Button";
+import { Command, CommandGroup, CommandItem, CommandList } from "../../../../components/ui/Command";
+import { Form, FormControl, FormField, FormItem, FormMessage } from "../../../../components/ui/Form";
+import { Input } from "../../../../components/ui/Input";
+import mudLogo from "../../icon.svg";
+import { getWorldUrl } from "../../utils/getWorldUrl";
+
+const formSchema = z.object({
+  worldAddress: z
+    .string()
+    .refine((value) => isAddress(value), {
+      message: "Invalid world address",
+    })
+    .transform((value): Address => value as Address),
+});
+
+export function WorldsForm({ worlds }: { worlds: Address[] }) {
+  const router = useRouter();
+  const { chainName } = useParams();
+  const [open, setOpen] = useState(false);
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    reValidateMode: "onChange",
+  });
+
+  function onSubmit({ worldAddress }: z.infer<typeof formSchema>) {
+    router.push(getWorldUrl(chainName as string, worldAddress));
+  }
+
+  function onLuckyWorld() {
+    if (worlds.length > 0) {
+      const luckyAddress = worlds[Math.floor(Math.random() * worlds.length)];
+      router.push(getWorldUrl(chainName as string, luckyAddress));
+    }
+  }
+
+  return (
+    <div className="mx-auto flex min-h-screen w-[450px] flex-col items-center justify-center p-4">
+      <h1 className="mb-6 flex items-center gap-4 self-start font-mono text-4xl font-bold uppercase">
```

Comment:
wonder if we should either center the logo+title in the container or center the title in the remaining space after the left-aligned logo

Reply:
tried both ways but somehow the current left-aligned version still looks best to me
![CleanShot 2024-10-08 at 15 08 47@2x](https://github.com/user-attachments/assets/eed2905f-d042-4998-ab6a-53d59a74ea39)

Reply:
I think I prefer either of those to the current state, with the last one being my favorite